examples/extsort/main.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	cMaxRandomNum = 10000
+)
+
+func main() {
+	var numDataElements, maxMemSortSize int
+	var inputPath, tempPath, outputPath string
+	flag.IntVar(&numDataElements, "num", 1000, "# elements to generate & sort")
+	flag.IntVar(&maxMemSortSize, "nummem", 100, "# elements to sort in mem")
+	flag.StringVar(&inputPath, "i", "input.dat", "input file (overwritten)")
+	flag.StringVar(&tempPath, "t", "bq", "path to write intermediate data")
+	flag.StringVar(&outputPath, "o", "output.dat", "output file (overwritten)")
+	flag.Parse()
+
+	// generate random data for testing
+	if err := generateData(numDataElements, cMaxRandomNum, inputPath); err != nil {
+		panic(err)
+	}
+
+	// sort the data
+	if err := ExternalSort(inputPath, tempPath, outputPath, maxMemSortSize); err != nil {
+		panic(err)
+	}
```

Comment:
After this step we should add a validator to check if sorting was correct or not. It will just read elements from output file and compare last two read elements.

Reply:
added a validation step